### PR TITLE
[Feat] Stroke color for Slider value indicator

### LINKED ISSUE #135984: [A11Y] More contrast in Slider widget value indicator
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Use case

Consider a scenario where the background color and indicator's background color are the same. 

Adding a stroke color to the value indicator would be a valuable for the following reasons:
- **Visual Clarity:** It would allow developers to make the value indicator stand out more against the background, making it easier for users to notice.
- **Customization:** It would provide more flexibility in customizing the appearance of the sliding widget, allowing developers to match the design requirements of their apps.
- **Accessibility:** Improved visual distinction can enhance the accessibility of the sliding widget for users with various needs.

**Current**
current

### Proposal

expected

### PR DESCRIPTION
**Pull request by 838**
Consider a scenario where the background color and indicator's background color are the same. 

Adding a stroke color to the value indicator would be a valuable for the following reasons:
- **Visual Clarity:** It would allow developers to make the value indicator stand out more against the background, making it easier for users to notice.
- **Customization:** It would provide more flexibility in customizing the appearance of the sliding widget, allowing developers to match the design requirements of their apps.
- **Accessibility:** Improved visual distinction can enhance the accessibility of the sliding widget for users with various needs.

*List which issues are fixed by this PR. You must list at least one issue.*

Fixes #135984

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by TahaTesser** [comment]
@838 
Thanks for contributing!

Overall this looks great. One should've one test that checks the Slider using `RectangularSliderValueIndicatorShape` with the new `valueIndicatorStrokeColor` value indicator is actually rendered as expected.

**Comment by 838** [comment]
@TahaTesser 
Thanks for the feedback 😊.

The test is now added.

File: packages/flutter/test/material/slider_test.dart
Comment (TahaTesser) [inline packages/flutter/test/material/slider_test.dart:2703]: This is a good start.  However, we're testing a `SliderTheme` property for a partuclar value indicator shape and the test should in the `slider_theme_test.dart` class.

Here I simplified the test, remove unrelated code and renamed the test. You can remove the `slider_test.dart` test.

Please place this in the `slider_theme_test.dart` after line `2086` and before `group('Material 2', () {` 
```dart
  testWidgets('RectangularSliderValueIndicatorShape supports SliderTheme.valueIndicatorStrokeColor', (WidgetTester tester) async {
    final ThemeData theme = ThemeData(
      sliderTheme: const SliderThemeData(
        showValueIndicator: ShowValueIndicator.always,
        valueIndicatorShape: RectangularSliderValueIndicatorShape(),
        valueIndicatorColor: Color(0xff000001),
        valueIndicatorStrokeColor: Color(0xff000002),
      ),
    );

    const double value = 0.5;

    await tester.pumpWidget(
      MaterialApp(
        theme: theme,
        home: Material(
          child: Center(
            child: Slider(
              value: value,
              label: '$value',
              onChanged: (double newValue) {},
            ),
          ),
        ),
      ),
    );

    final RenderBox valueIndicatorBox = tester.renderObject(find.byType(Overlay));

    final Offset center = tester.getCenter(find.byType(Slider));
    await tester.startGesture(center);
    // Wait for value indicator animation to finish.
    await tester.pumpAndSettle();

    expect(
      valueIndicatorBox,
      paints
        ..path(color: (theme.colorScheme.shadow)) // shadow
        ..path(color: theme.sliderTheme.valueIndicatorStrokeColor)
        ..path(color: theme.sliderTheme.valueIndicatorColor),
    );
  });
```

**Review by TahaTesser — CHANGES_REQUESTED** [review]

File: packages/flutter/test/material/slider_test.dart
Comment (838) [reply packages/flutter/test/material/slider_test.dart]: Thank you for your feedback and guidance! I appreciate your input. 

I've made the necessary changes :).

File: packages/flutter/lib/src/material/slider_theme.dart
Comment (TahaTesser) [inline packages/flutter/lib/src/material/slider_theme.dart:428]:
```suggestion
  /// The color given to the [valueIndicatorShape] stroke.
```

**Review by TahaTesser — CHANGES_REQUESTED** [review]

**Review by TahaTesser — APPROVED** [review]
LGTM! We'll need a second review from @Piinks.

File: packages/flutter/lib/src/material/slider_theme.dart
Comment (Piinks) [inline packages/flutter/lib/src/material/slider_theme.dart:429]: Looking at the related property above, `valueIndicatorColor`, that applies to the `RectangularSliverValueIndicatorShape`, but also:
- `PaddleSliderValueIndicatorShape`
- `PaddleRangeSliverValueIndicatorShape`
- `DropSliderValueIndicatorShape`

Should these other classes also use this new property? I think they should, but if not for some reason, we should probably document the difference in how the new property is applied as folks might expect it to be the same as the other property.

**Review by Piinks — COMMENTED** [review]
Hi @838 welcome! Thank you for contributing.

File: packages/flutter/lib/src/material/slider_theme.dart
Comment (838) [reply packages/flutter/lib/src/material/slider_theme.dart:429]: HI @Piinks

Thanks for your feedback 🙂. 

I completely agree with you! It makes a lot of sense to maintain consistency across these related classes.

I'll make sure to explore the possibility of extending the use of the "valueIndicatorStorkeColor" property to the other classes you mentioned.

**Comment by TahaTesser** [comment]
@838 
Looks like the changes are failing some tests in the `slider_theme_test.dart` class

File: packages/flutter/lib/src/material/slider_theme.dart
Comment (838) [inline packages/flutter/lib/src/material/slider_theme.dart:3563]: @TahaTesser 
Are these modifications accurate for incorporating stroke colors into the DropSliderValueIndicatorShape?

Comment (TahaTesser) [reply packages/flutter/lib/src/material/slider_theme.dart:3563]: Looks good

File: packages/flutter/lib/src/material/slider_theme.dart
Comment (TahaTesser) [inline packages/flutter/lib/src/material/slider_theme.dart:2717]: Looks like you added tests for all the value indicator shapes except  this `RectangularRangeSliderValueIndicatorShape`

File: packages/flutter/test/material/slider_theme_test.dart
Comment (TahaTesser) [inline packages/flutter/test/material/slider_theme_test.dart:2224]: `RectangularRangeSliderValueIndicatorShape` test here

**Review by TahaTesser — CHANGES_REQUESTED** [review]
Good job with the updates.

While reviewing your PR, I've created code sample to check all the shapes you've updated here

### Code sample

<details>
<summary>expand to view the code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      theme: ThemeData.dark(useMaterial3: true),
      home: const Example(),
    );
  }
}

class Example extends StatefulWidget {
  const Example({super.key});

  @override
  State<Example> createState() => _ExampleState();
}

class _ExampleState extends State<Example> {
  double _sliderValue = 0.5;
  RangeValues _rangeSliderValues = const RangeValues(0.25, 0.75);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Sliders'),
      ),
      body: Column(
        mainAxisAlignment: MainAxisAlignment.spaceEvenly,
        children: [
          _buildSlider(range: false),
          _buildSlider(
            range: false,
            rectangularSliderValueIndicatorShape:
                const RectangularSliderValueIndicatorShape(),
          ),
          _buildSlider(
            range: false,
            paddleSliderValueIndicatorShape:
                const PaddleSliderValueIndicatorShape(),
          ),
          _buildSlider(
            range: true,
            rectangularRangeSliderValueIndicatorShape:
                const RectangularRangeSliderValueIndicatorShape(),
          ),
          _buildSlider(
            range: true,
            paddleRangeSliderValueIndicatorShape:
                const PaddleRangeSliderValueIndicatorShape(),
          ),
        ],
      ),
    );
  }

  Widget _buildSlider({
    bool range = false,
    RectangularSliderValueIndicatorShape? rectangularSliderValueIndicatorShape,
    PaddleSliderValueIndicatorShape? paddleSliderValueIndicatorShape,
    RectangularRangeSliderValueIndicatorShape? rectangularRangeSliderValueIndicatorShape,
    PaddleRangeSliderValueIndicatorShape? paddleRangeSliderValueIndicatorShape,
  }) {
    if (range) {
      return SliderTheme(
        data: SliderThemeData(
          rangeValueIndicatorShape: rectangularRangeSliderValueIndicatorShape ??
              paddleRangeSliderValueIndicatorShape,
          valueIndicatorColor: Theme.of(context).colorScheme.surface,
          valueIndicatorStrokeColor: Theme.of(context).colorScheme.tertiary,
          showValueIndicator: ShowValueIndicator.always,
          valueIndicatorTextStyle: TextStyle(
            color: Theme.of(context).colorScheme.onSurface,
          ),
        ),
        child: RangeSlider(
          values: _rangeSliderValues,
          onChanged: (value) {
            setState(() {
              _rangeSliderValues = value;
            });
          },
          labels: RangeLabels(
            _rangeSliderValues.start.round().toString(),
            _rangeSliderValues.end.round().toString(),
          ),
          min: 0,
          max: 1,
        ),
      );
    }
    return SliderTheme(
      data: SliderThemeData(
        valueIndicatorShape: rectangularSliderValueIndicatorShape ??
            paddleSliderValueIndicatorShape,
        valueIndicatorColor: Theme.of(context).colorScheme.surface,
        valueIndicatorStrokeColor: Theme.of(context).colorScheme.tertiary,
        showValueIndicator: ShowValueIndicator.always,
        valueIndicatorTextStyle: TextStyle(
          color: Theme.of(context).colorScheme.onSurface,
        ),
      ),
      child: Slider(
        value: _sliderValue,
        onChanged: (value) {
          setState(() {
            _sliderValue = value;
          });
        },
        label: _sliderValue.round().toString(),
        min: 0,
        max: 1,
      ),
    );
  }
}
```

</details>

**Comment by 838** [comment]
@TahaTesser 
Should I also add a test to verify that 'overlappingShapeStrokeColor' behaves correctly when a `valueIndicatorStrokeColor` is provided for `RangeSlider`?

**Comment by TahaTesser** [comment]
Yes, in the same test class

File: packages/flutter/test/material/slider_theme_test.dart
Comment (TahaTesser) [inline packages/flutter/test/material/slider_theme_test.dart:2277]: Is this a duplicate test?

File: packages/flutter/test/material/slider_theme_test.dart
Comment (TahaTesser) [inline packages/flutter/test/material/slider_theme_test.dart:2384]: This looks like a duplicate too

**Review by TahaTesser — CHANGES_REQUESTED** [review]

File: packages/flutter/test/material/slider_theme_test.dart
Comment (838) [reply packages/flutter/test/material/slider_theme_test.dart:2384]: Same as [this comment](https://github.com/flutter/flutter/pull/135986/files#r1363503654) but instead testing `PaddleRangeSliderValueIndicatorShape`

File: packages/flutter/test/material/slider_theme_test.dart
Comment (TahaTesser) [reply packages/flutter/test/material/slider_theme_test.dart:2277]: Good idea

**Review by TahaTesser — APPROVED** [review]
LGTM!

We'll need a second review before we land this.

File: packages/flutter/test/material/slider_theme_test.dart
Comment (TahaTesser) [reply packages/flutter/test/material/slider_theme_test.dart:2277]: Good idea

**Comment by Piinks** [comment]
Goldens are approved!

**Review by Piinks — APPROVED** [review]
Flutter_LGTM
Thank you for the contribution!

**Pull request merged**
